serialization/chrome_pak.ksy

```diff
@@ -0,0 +1,90 @@
+meta:
+  id: chrome_pak
+  title: Chrome PAK serialization format
+  file-extension: pak
+  license: CC0-1.0
+  endian: le
+doc: |
+  Format mostly used by Google Chrome and various Android apps to store
+  resources such as translated strings, help messages and images.
+doc-ref:
+  - http://dev.chromium.org/developers/design-documents/linuxresourcesandlocalizedstrings # version 4
+  - https://chromium.googlesource.com/chromium/src/tools/grit/+/22f7a68bb5ad68fe4192d0f34466049038735b9c/grit/format/data_pack.py # version 4
+  - https://chromium.googlesource.com/chromium/src/tools/grit/+/c1a76d6143016e9b2c292236bab623d1497cd31f/grit/format/data_pack.py # version 5
+seq:
+  - id: version
+    type: u4
+    doc: Support versions are 4 and 5
+  - id: header
+    type:
+      switch-on: version
+      cases:
+        4: header4
+        5: header5
+types:
+  header4:
+    seq:
+      - id: number_of_resources
+        type: u4
+      - id: encoding
+        type: u1
+        enum: encoding
+      - id: resources
+        type: resource
+        repeat: expr
+        repeat-expr: number_of_resources
+      - id: end_of_resource
+        type: end_of_resource
+        doc: |
+          The length is calculated by looking at the offset of
+          the next item, so an extra entry is stored with id 0
+          and offset pointing to the end of the resources.
+  header5:
+    seq:
+      - id: encoding
+        type: u1
+        enum: encoding
+      - id: ignore
+        size: 3
+        doc: These three bytes have no meaning in the file
+      - id: number_of_resources
+        type: u2
+      - id: number_of_aliases
+        type: u2
+      - id: resources
+        type: resource
+        repeat: expr
+        repeat-expr: number_of_resources
+      - id: end_of_resource
+        type: end_of_resource
+        doc: |
+          The length is calculated by looking at the offset of
+          the next item, so an extra entry is stored with id 0
+          and offset pointing to the end of the resources.
+      - id: aliases
+        type: alias
+        repeat: expr
+        repeat-expr: number_of_aliases
+  alias:
+    seq:
+      - id: id
+        type: u2
+      - id: index
+        type: u2
+  resource:
+    seq:
+      - id: id
+        type: u2
+      - id: offset
+        type: u4
+  end_of_resource:
```

Comment:
```yaml
seq:
  - id: end_resource
    type: resource
    valid:
      expr: _.id == 0
```

Reply:
That is probably better. Thanks!

Reply:
No, it is not. Just the array should have `number_of_resources + 1` items. And the actual array of resources should have `number_of_resources` items, then you can safely use `[_index + 1]` to calculate length.

Reply:
Luckily I haven't changed it yet. As said, making the array bigger makes more sense, but I want to make sure that the index of the last is 0.

Reply:
Infortunately we don't have `valid` for `value` `instance`s. But the check still can be done in `valid` in some other place than the field itself. So you need to validate that the last element id is 0.